### ISSUE TITLE
odbc freetds could not permit subquerys?

### ISSUE THREAD

**Issue by mckaygerhard** [FreeTDS/freetds#131]
i found a old (or new depends if newer driver works for me) bug.. 

odbc does not permit to me made subquerys... i mean.. 
a) made a quiery with current conection
b) iterate on results
c) try another query with current conection but not finish to iterate
d) get a error from low level (i mean from odbc modulo, that are freetds)

the programing languajes are gambas and visual basic 6 same error on both, from php too but in php i not tested deeply, shoul that fill in new issue?

**Comment by freddy77** [comment]
This is a protocol limitation. Workarounds are cursors and MARS. In ODBC the terminology is multiple active statements.

**Comment by mckaygerhard** [comment]
hi, freddy i think you dont understand the proble, i assume that the odbc implementation offers the cursor right? and the querys itarate over the cursor! so i was thinking that the code implementation in gabmas or/and visualbasic are using always the cursor!

**Comment by freddy77** [comment]
Is not automatic, depends on the type of cursor they use. The default type are not enough. So it implementation dependent. You can enable MARS (on the connection) and use ODBC.

**Comment by mckaygerhard** [comment]
hi @freddy77, remenber that i dont taking about the Mocosoft SQLserver, i'm using sybase.. does handle the ODBC freetds those types of cursos respect sybase feaured? does sybase have the usage of that "mars" feature? i guess now i made the right questions and depends of the response this can be a bug or not

**Comment by freddy77** [comment]
Yes, you are right, cursors are not supported for Sybase.

**Comment by mckaygerhard** [comment]
umm its hard to believe! maybe we are poor informated about it! must be a workaround or something that we cannot see it!

**Comment by mckaygerhard** [comment]
umm seems here http://sqlanywhere-forum.sap.com/questions/30539/does-sybase-support-something-like-mars-cursors i gues that its not supported but i ask directly now to a trith partner of sybase.. i'll waith the answerd and pot feedback here..

**Comment by mckaygerhard** [comment]
so based on http://sqlanywhere-forum.sap.com/questions/30539/does-sybase-support-something-like-mars-cursors i'll close that.. i change the issue title to point clarify

**Comment by mckaygerhard** [comment]
i reopened due that:
> Currently, the only way to enable MAS in FreeTDS is to use a dynamic cursor.

what about it?

**Comment by mckaygerhard** [comment]
tested with odbc provided by easysoft and java also .net, here a code in java:

easysoft said that must be enabled dynamic cursors, how?
```
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;
import sybase.jdbc4.sqlanywhere.*;

public class marsOdbc {

	public static void main(String[] args) throws Exception   {
		// TODO Auto-generated method stu
		String query = "select * from EXAMPLE0001";
		Connection conex = DriverManager.getConnection("jdbc:sqlanywhere:uid=DBA;pwd=sql;dsn=hablador");
		Statement stmt = conex.createStatement();
		Statement stmt2 = conex.createStatement();
		Statement stmt3 = conex.createStatement();
		ResultSet rs=stmt.executeQuery(query);
		ResultSet rs2;
		while (rs.next()){
		System.out.println(rs.getInt("a") + "   " + rs.getString("b"));	
		stmt2.executeUpdate("update EXAMPLE0001 set b='Cam.Nuevo' where a="+rs.getInt("a") );
		rs2=stmt3.executeQuery("select * from tc_codmsc");
		    while(rs2.next()) {
		    	System.out.println(rs.getString(1));			    	
		    }
		}
		rs=stmt.executeQuery(query);
		while (rs.next()){
			System.out.println(rs.getInt("a") + "   " + rs.getString("b"));	
			
			}
		conex.close();

	}

}
```

**Comment by freddy77** [comment]
Yes, currently cursors are not supported under Sybase (protocol 5.0). The only workaround is use multiple connections.

**Comment by mckaygerhard** [comment]
hi @freddy77, i think its supported, in java with same connection (really hate java) i can retrieve data.. but in linux using freetds cannot do that! i think its a freetds limitation! a partner in my job can doit with same code in java.. but in freetds in same guindows OS cannot! so i think could check that a little...

also the private env that i promised i'll send to you later..

**Comment by freddy77** [comment]
What I was trying to say is that we (FreeTDS) do not support cursors using Sybase server. Yes, Sybase supports them.

**Comment by mckaygerhard** [comment]
arrgggg, its so hard to implementing at short time right? i must let open this issue or close it? (based on your response), if let open i must change tittle to "feature request"?

**Comment by freddy77** [comment]
Yes, should go in the feature request. My big item todo list is:
- finish Sybase 16 with new calls;
- release;
- encrypted fields.

**Comment by mckaygerhard** [comment]
i have sybase 16 instalation (16 only develper license, 12 complete license) for tests.. 

for access to environment sorry for late, i'll still busy but next week i have somenthing

now i changed title, and please reamin low dependency (for my embebed devices with older linuxes)
